### PR TITLE
Added More Functionality

### PR DESCRIPTION
**Pull request by IamFaizanKhalid**
- Separate Routes File
- Added  Logging
- Allowed `PATCH` and `DELETE`
- Common Authentication for All APIs
- Added `.env`
- Dockerized Project
- Created Go Mod

### CODE REVIEW

File: apis/auth.go [hunk 1]
```diff
@@ -1,52 +1,117 @@
...
-func verifyUserInfo(user models.SignInRequest) models.User {
+func verifyUserInfo(user models.SignInRequest) (models.User, error) {
 	switch {
 	case user.Username == "admin" && user.Password == "admin":
-		return models.User{user.Username, models.USER_ADMIN}
+		return models.User{
+			Name: user.Username,
+			Role: models.USER_ADMIN,
+		}, nil
 	case user.Username == "faizan" && user.Password == "faizan":
```
Comment (fahadsiddiqui) [inline apis/auth.go:52]: "faizan"? Change this to "john doe" https://en.wikipedia.org/wiki/John_Doe.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      - HTTP_PORT=${HTTP_PORT}
+      - JWT_SIGN_KEY=${JWT_SIGN_KEY}
+      - JWT_VERIFY_KEY=${JWT_VERIFY_KEY}
+      - DB_USER=${DB_USER}
+      - DB_NAME=${DB_NAME}
+      - DB_PASSWORD=${DB_PASSWORD}
+      - DB_TZ=${DB_TZ}
+      - DB_SSL_MODE=${DB_SSL_MODE}
+    ports:
+      - ${HTTP_PORT}:${HTTP_PORT}
```
Comment (fahadsiddiqui) [inline docker-compose.yml:18]: Do this is Dockerfile also. Rather `EXPOSE 6969` there, there should be env var somehow propagated.

**Review by fahadsiddiqui — CHANGES_REQUESTED** [review]
Minor changes.

File: apis/auth.go (hunk 1 above)
Comment (IamFaizanKhalid) [reply apis/auth.go]: Done.. :smile: :cry:

File: docker-compose.yml (hunk 2 above)
Comment (IamFaizanKhalid) [reply docker-compose.yml:18]: Done..

**Pull request merged**